plugins/confluent/confluent.go

```diff
@@ -0,0 +1,27 @@
+package confluent
+
+import (
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/needsauth"
+	"github.com/1Password/shell-plugins/sdk/schema"
+)
+
+func ConfluentCLI() schema.Executable {
+	return schema.Executable{
+		Name:    "Confluent CLI",
+		Runs:    []string{"confluent"},
+		DocsURL: sdk.URL("https://docs.confluent.io/confluent-cli/current/overview.html"),
+		NeedsAuth: needsauth.IfAll(
+			needsauth.ForCommand("login"),
```

Comment:
Does the `login` command write credentials to disk to pass them to the other commands? One of the goals of Shell Plugins is to avoid credentials ever having to touch the disk in plaintext. Instead they're stored encrypted in 1Password and are only passed to the tools that need the credential in memory and only to authorized terminal sessions.

In case Confluent uses a file to pass the credentials between the login and other commands, we can use the [file-provisioner](https://github.com/1Password/shell-plugins/blob/main/sdk/provision/file_provisioner.go) to pass the crednetials directly to the commands that require them (without needing to run `login` beforehand). It uses a temporary in-memory file to pass the credentials to the CLI. As a reference example, a.o. the [MySQL plugin use a temporary file to pass the credential](https://github.com/1Password/shell-plugins/blob/e6148953ffd3a24a79e3fa4eb513fa8bd44d96a5/plugins/mysql/database_credentials.go#L46).

Let us know if we can help out. In addition to here on GitHub we're also on [Slack](https://join.slack.com/t/1password-devs/shared_invite/zt-1halo11ps-6o9pEv96xZ3LtX_VE0fJQA) for quick back-and-forth and you can book a [pairing session](https://calendly.com/d/grs-x2h-pmb/1password-shell-plugins-pairing-session) with us.

Reply:
The login command performs the exchange of credentials for a JWT Token that has a TTL (Default 1hr if I recall). So without implementing the login API calls within the OP Plugin itself and writing out the token to a temp file. This was the only way I could see fit to connect the 2. As credentials aren't able to be provide in any other way at this time.

It also worth adding that the APIs themselves are rate limited. So even attempting to login, get the token, write it to disk, then performance the user intended action. Would be problematic for the end power user.

Reply:
> It also worth adding that the APIs themselves are rate limited. So even attempting to login, get the token, write it to disk, then performance the user intended action. Would be problematic for the end power user.

For this the plugin ecosystem offers an [encrypted cache](https://github.com/1Password/shell-plugins/blob/e6148953ffd3a24a79e3fa4eb513fa8bd44d96a5/sdk/provisioner.go#L169) to persist the token between command executions. An example of this in use are [temporary AWS credentials](https://github.com/1Password/shell-plugins/blob/e6148953ffd3a24a79e3fa4eb513fa8bd44d96a5/plugins/aws/sts_provisioner.go#L67).

> So without implementing the login API calls within the OP Plugin itself and writing out the token to a temp file.

Is there a Confluent SDK we could use? Or could we possibly capture the output of the `login` command before it hits the disk?

Reply:
As it stands, there currently is no way to provide the token to the CLI. So while there is a GoLang SDK, there wouldn't be a way at this time to then translate that token to the CLI. I will open a separate ticket/issue with the Confluent CLI team and see about addressing this at a later date. But for now this seems to be the only route forward.

Reply:
A possible reason would be that this code currently lives in the `internal` dir (see https://github.com/golang-standards/project-layout#internal). 🤔 

@nerdynick what would be the way to go about asking for some changes in the Confluent CLI's codebase, if we were to go down this route?

Reply:
I'd be slightly hesitant about attempting to manipulate the internal state cache that the CLI manages. It's constantly changing without backward compatibility, since it's deemed ephemeral and internal to the CLI itself and would require keeping up with new releases, which happen regularly. Outside of going the route that I did with this PR so far. Would be to file an issue against the project or do the work ourselves and hope a PR gets accepted to extend support for a Token to be provided via ENV Vars like the User/Pass can be. For which I started exploring down that route when this thread was brought up. It'd take a bit of time to complete and then also hope to get the PR accepted. 

The follow on question would then be, if we go down the route of asking the Confluent CLI team to make the needed change, would this project, 1Pass CLI Plugins, be OK with importing an external project (SDK) to acquire that token or would we need to duplicate that logic in a minimal fashion within this plugin to achieve that?

Reply:
>The follow on question would then be, if we go down the route of asking the Confluent CLI team to make the needed change, would this project, 1Pass CLI Plugins, be OK with importing an external project (SDK) to acquire that token or would we need to duplicate that logic in a minimal fashion within this plugin to achieve that?

The best way forward here seems to start work on the Confluent CLI to be able to take in the token via the env var, as you said. After this is done, we are more than open to import any external SDK that would retrieve that token.

Reply:
Alright, I was able to modify the plugin and make use of what is essentially an undocumented feature. In that the Confluent CLI is able to take the credentials on every command. So the plugin now provides credentials on all calls but the Help and Version ones. This should mitigate the security concern that have been brought up I believe. This should also be a future proof route as the API documents the use of Credentials on all calls rather then the use of a Token.

Reply:
Hi @nerdynick !
I was just trying to use this shell plugin but can't get the username + password to work with commands without first running `confluent login` and writing an auth_token to disk. Is there a trick to get this to work without that step? 

```➜  ~ confluent kafka topic list --url https://<my-domain>.us-east-2.aws.confluent.cloud:443
No session token found, please enter user credentials. To avoid being prompted, run `confluent login`.
Username:```